components/core/src/clp/clo/clo.cpp

```diff
@@ -91,6 +99,244 @@ static bool search_archive(
+                    e.get_line_number(),
+                    e.what(),
+                    error_code
+            );
+        }
+        return_value = false;
+    }
+    return return_value;
+}
+
+/**
+ * Extracts a file split as IR chunks, writing them to the local filesystem and writing their
+ * metadata to the results cache.
+ * @param command_line_args
+ * @return Whether the file split was successfully extracted.
+ */
+bool extract_ir(CommandLineArguments const& command_line_args) {
+    auto const archive_path{std::filesystem::path(command_line_args.get_archive_path())};
+    if (false == std::filesystem::exists(archive_path)) {
+        SPDLOG_ERROR("Archive '{}' doesn't exist.", archive_path.c_str());
+        return false;
+    }
+    auto archive_metadata_file = archive_path / clp::streaming_archive::cMetadataFileName;
+    if (false == std::filesystem::exists(archive_metadata_file)) {
+        SPDLOG_ERROR(
+                "Archive metadata file '{}' does not exist. '{}' may not be an archive.",
+                archive_metadata_file.c_str(),
+                archive_path.c_str()
+        );
+        return false;
+    }
+
+    try {
+        // Create output directory in case it doesn't exist
+        auto const output_dir{boost::filesystem::path(command_line_args.get_ir_output_dir())};
+        if (auto error_code = create_directory(output_dir.parent_path().string(), 0700, true);
+            ErrorCode_Success != error_code)
+        {
+            SPDLOG_ERROR(
+                    "Failed to create {} - {}",
+                    output_dir.parent_path().c_str(),
+                    strerror(errno)
+            );
+            return false;
+        }
+
+        Archive archive_reader;
+        archive_reader.open(archive_path.string());
+        archive_reader.refresh_dictionaries();
+
+        auto const& file_split_id = command_line_args.get_file_split_id();
+        auto file_metadata_ix_ptr = archive_reader.get_file_iterator_by_split_id(file_split_id);
+        if (false == file_metadata_ix_ptr->has_next()) {
+            SPDLOG_ERROR(
+                    "File split doesn't exist {} in the archive {}",
+                    file_split_id,
+                    archive_path.string()
+            );
+            return false;
+        }
+
+        mongocxx::client client;
+        mongocxx::collection collection;
+        std::vector<bsoncxx::document::value> results;
+
+        try {
+            auto const mongo_uri{mongocxx::uri(command_line_args.get_ir_mongodb_uri())};
+            client = mongocxx::client{mongo_uri};
+            collection
+                    = client[mongo_uri.database()][command_line_args.get_ir_mongodb_collection()];
+        } catch (mongocxx::exception const& e) {
+            SPDLOG_ERROR("Failed to connect to results cache - {}", e.what());
+            return false;
+        }
+
+        auto ir_output_handler = [&](boost::filesystem::path const& src_ir_path,
```

Comment:
Since only `result` is used, shall we capture `result` only?

Reply:
@kirkrodrigues 
Shall we do explicit capture?

Reply:
I think relying on the compiler to figure out what's used is easier. So I would opt for &